src/ws/remotectl-certificate.c

```diff
@@ -149,12 +140,129 @@ ensure_certificate (const gchar *user,
   ret = 0;
 
 out:
+  g_clear_error (&error);
+  return ret;
+}
+
+static int
+ensure_certificate (const gchar *user,
+                    const gchar *group,
+                    const gchar *selinux)
+{
+  GError *error = NULL;
+  GTlsCertificate *certificate = NULL;
+  gchar *path = NULL;
+  gint ret = 1;
+
+  path = cockpit_certificate_locate (TRUE, &error);
+  if (path != NULL)
+    certificate = cockpit_certificate_load (path, &error);
+
+  if (!certificate)
+    {
+      g_message ("%s", error->message);
+      goto out;
+    }
+
+  ret = set_cert_attributes (path, user, group, selinux);
+
+out:
   g_clear_object (&certificate);
   g_clear_error (&error);
   g_free (path);
   return ret;
 }
 
+static gint
+cockpit_certificate_combine (gchar **pem_files,
+                             const gchar *user,
+                             const gchar *group,
+                             const gchar *selinux)
+{
+  const gchar * const* dirs = cockpit_conf_get_dirs ();
+
+  GError *error = NULL;
+  GTlsCertificate *certificate = NULL;
+  GString *combined = g_string_new ("");
+
+  gchar *cert_dir = NULL;
+  gchar *name = NULL;
+  gchar *spot = NULL;
+
+  gchar *target_name = NULL;
+  gchar *target_path = NULL;
+
+  gint ret = 1;
+  gint i;
+
+  cert_dir = g_build_filename (dirs[0], "cockpit", "ws-certs.d", NULL);
+  if (g_mkdir_with_parents (cert_dir, 0700) != 0)
+    {
+      g_message ("Error creating directory %s: %m", cert_dir);
+      goto out;
+    }
+
+  /* target file is named after the first file */
+  name = g_path_get_basename (pem_files[0]);
+  spot = strrchr (name, '.');
+  if (spot != NULL)
+    *spot = '\0';
+
+  target_name = g_strdup_printf ("%s.cert", name);
+  target_path = g_build_filename (cert_dir, target_name, NULL);
+
+  for (i = 0; pem_files[i] != NULL; i++)
+    {
+      gchar *data = NULL;
+      gsize length;
+      gboolean success = g_file_get_contents (pem_files[i], &data, &length, &error);
+
+      if (success)
+        g_string_append_printf (combined, "%s\n", data);
+
+      g_free (data);
+
+      if (!success)
+        goto out;
+    }
+
+  if (!g_file_set_contents (target_path, combined->str, combined->len, &error))
```

Comment:
Overwriting is fine. Will also be the case on certificate renewal. Lets just document it properly.